src/openloco/windows/about_music.cpp

```diff
@@ -0,0 +1,187 @@
+#include "../graphics/colours.h"
+#include "../graphics/gfx.h"
+#include "../interop/interop.hpp"
+#include "../localisation/string_ids.h"
+#include "../objects/interface_skin_object.h"
+#include "../objects/objectmgr.h"
+#include "../windowmgr.h"
+
+using namespace openloco::interop;
+
+namespace openloco::ui::about_music
+{
+    constexpr uint16_t ww = 500;
+    constexpr uint16_t wh = 312;
+
+    namespace widx
+    {
+        enum
+        {
+            frame,
+            title,
+            close,
+            panel,
+            scrollview,
+        };
+    }
+
+    static widget_t _widgets[] = {
+        make_widget({ 0, 0 }, { ww, wh }, widget_type::frame, 0),
+        make_widget({ 1, 1 }, { ww - 2, 13 }, widget_type::caption_25, 0, string_ids::music_acknowledgements_caption),
+        make_widget({ ww - 15, 2 }, { 13, 13 }, widget_type::wt_9, 0, 2321, string_ids::tooltip_close_window),
+        make_widget({ 0, 15 }, { ww, 297 }, widget_type::panel, 1),
+        make_widget({ 4, 18 }, { ww - 8, 289 }, widget_type::scrollview, 1, 2),
+        widget_end(),
+    };
+
+    static window_event_list _events;
+
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex);
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight);
+    static void tooltip(ui::window* window, widget_index widgetIndex);
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex);
+
+    // 0x0043B26C
+    void open()
+    {
+        if (windowmgr::bring_to_front(window_type::about_music, 0) != nullptr)
+            return;
+
+        _events.on_mouse_up = on_mouse_up;
+        _events.get_scroll_size = get_scroll_size;
+        _events.tooltip = tooltip;
+        _events.draw = draw;
+        _events.draw_scroll = draw_scroll;
+
+        auto window = windowmgr::create_window_centred(
+            window_type::about_music,
+            ww,
+            wh,
+            0,
+            &_events);
+
+        window->widgets = _widgets;
+        window->enabled_widgets = (1 << widx::close) | (1 << widx::scrollview);
+        window->init_scroll_widgets();
+
+        auto interface = objectmgr::get<interface_skin_object>();
+        window->colours[0] = interface->colour_0B;
+        window->colours[1] = interface->colour_10;
+    }
+
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close:
+                windowmgr::close(window->type);
+                break;
+        }
+    }
+
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+    {
+        *scrollHeight = 1054;
```

Comment:
Can you calculate this based on `count(strings_to_draw)`?

Reply:
Sure, but I was hoping to avoid exposing the list of strings to the entire namespace to allow the compiler to optimise it away. Perhaps it's best to give the list a size constant rather than having the compiler infer its size, which we can then use to compute the height.

Reply:
It's an about window that will rarely get opened. I sincerely doubt whether any speed optimisation is worth it.